freqtrade/freqai/data_kitchen.py

```diff
@@ -287,19 +287,26 @@ def normalize_data(self, data_dictionary: Dict) -> Dict[Any, Any]:
         :returns:
         :data_dictionary: updated dictionary with standardized values.
         """
+
+        df_train_features = data_dictionary["train_features"]
         # standardize the data by training stats
-        train_max = data_dictionary["train_features"].max()
-        train_min = data_dictionary["train_features"].min()
-        data_dictionary["train_features"] = (
-            2 * (data_dictionary["train_features"] - train_min) / (train_max - train_min) - 1
+        train_max = df_train_features.max()
+        train_min = df_train_features.min()
+        df_train_features = (
+            2 * (df_train_features - train_min) / (train_max - train_min) - 1
         )
         data_dictionary["test_features"] = (
             2 * (data_dictionary["test_features"] - train_min) / (train_max - train_min) - 1
         )
 
         for item in train_max.keys():
-            self.data[item + "_max"] = train_max[item]
-            self.data[item + "_min"] = train_min[item]
+            if not [col for col in df_train_features.columns if col.startswith('PC')]:
+                self.data[item + "_max"] = train_max[item]
+                self.data[item + "_min"] = train_min[item]
+            else:
+                # if PCA is enabled and has transformed the training features
+                self.data[item + "_pca_max"] = train_max[item]
```

Comment:
`item` here is going to be unique already. In the case of principal components, `item` will be `PC1` `PC2` etc. Can't we just leave this code without the check then since the keys will be unique already in the coin meta data (`self.data`)?

Or is there another reason why you attach `_pca_max`, and I am missing it?

Reply:
You're right, the keys are already unique, my mistake!